### PR TITLE
Allow empty object for period, add toICALString method for duration

### PR DESCRIPTION
**Pull request by kewisch**
For my use case I do something like this:

``` javascript
var f = new ICAL.Period();
...
f.start = somestart;
f.end = someend;
```

This patch allows creating an empty ICAL.Period object. It also adds a getter for the End date thats independent of if duration or end was passed.

### CODE REVIEW

File: test/period_test.js [hunk 1]
```diff
@@ -26,12 +29,16 @@ suite('ical/period', function() {
         }
 
         if ('end' in data) {
-          assert.instanceOf(subject.end, ICAL.Time);
-          assert.hasProperties(
-            subject.end,
-            data.end,
-            'end property'
-          );
+          if (data['end']) {
```
Comment (lightsofapollo) [inline test/period_test.js]: nit: data.end

File: test/period_test.js [hunk 2]
```diff
@@ -55,6 +62,14 @@ suite('ical/period', function() {
           } 
           assert.hasProperties(dur, data.calculated_duration);
         }
+        if ('calculated_end' in data) {
```
Comment (lightsofapollo) [inline test/period_test.js]: nit: calculatedEnd

File: test/period_test.js [hunk 3]
```diff
@@ -55,6 +62,14 @@ suite('ical/period', function() {
...
+          var end = subject.getEnd();
+
+          if ('end' in data && data.end) {
+            assert.hasProperties(end, data.end, 'duration matches calculated');
+          } 
+          assert.hasProperties(end, data.calculated_end);
```
Comment (lightsofapollo) [inline test/period_test.js]: same nit as above

**Comment by lightsofapollo** [comment]
r=me with nits addressed.

File: test/period_test.js (hunk 1 above)
Comment (kewisch) [inline test/period_test.js]: This was copy/paste from data['duration']. I'll change both.

File: test/period_test.js (hunk 2 above)
Comment (kewisch) [inline test/period_test.js]: Same here, from calculated_duration

**Pull request merged**